src/tdydiscretization.c

```diff
@@ -0,0 +1,159 @@
+#include <private/tdycoreimpl.h>
+
+/* -------------------------------------------------------------------------- */
+/// Creates a PETSc global vector
+///
+/// @param [in] tdy     A TDy struct
+/// @param [out] vector A PETSc vector
+/// @returns 0 on success, or a non-zero error code on failure
+PetscErrorCode TDyCreateGlobalVector(TDy tdy, Vec *vector){
+
+  PetscFunctionBegin;
+  DM dm = tdy->dm;
+  PetscErrorCode ierr;
+
+  ierr = DMCreateGlobalVector(dm, vector); CHKERRQ(ierr);
+
+  PetscFunctionReturn(0);
+}
+
+/* -------------------------------------------------------------------------- */
+/// Creates a PETSc local vector
+///
+/// @param [in] tdy     A TDy struct
+/// @param [out] vector A PETSc vector
+/// @returns 0 on success, or a non-zero error code on failure
+PetscErrorCode TDyCreateLocalVector(TDy tdy, Vec *vector){
+
+  PetscFunctionBegin;
+  DM dm = tdy->dm;
+  PetscErrorCode ierr;
+
+  ierr = DMCreateLocalVector(dm, vector); CHKERRQ(ierr);
+
+  PetscFunctionReturn(0);
+}
+
+/* -------------------------------------------------------------------------- */
+/// Creates a PETSc natural vector
+///
+/// @param [in] tdy     A TDy struct
+/// @param [out] vector A PETSc vector
+/// @returns 0 on success, or a non-zero error code on failure
+PetscErrorCode TDyCreateNaturalVector(TDy tdy, Vec *vector){
+
+  PetscFunctionBegin;
+  PetscErrorCode ierr;
+
+  ierr = TDyCreateGlobalVector(tdy, vector); CHKERRQ(ierr);
+
+  PetscFunctionReturn(0);
+}
+
+/* -------------------------------------------------------------------------- */
+/// Duplicates a PETSc vector
+///
+/// @param [in] tdy      A TDy struct
+/// @param [in] vector1  A PETSc vector
+/// @param [out] vector2 A PETSc vector
+/// @returns 0 on success, or a non-zero error code on failure
+PetscErrorCode TDyDuplicateVector(TDy tdy, Vec vector1, Vec *vector2){
```

Comment:
+1

Reply:
Since we now have wrappers for creating various vectors (`TDyCreate*Vector`), I added this new function so we have a  consistent call to TDycore functions instead of PETSc functions directly.